test/e2e/performanceprofile/functests/utils/nodes/nodes.go

```diff
@@ -425,3 +426,68 @@ func GetByCpuCapacity(nodesList []corev1.Node, cpuQty int) []corev1.Node {
 	}
 	return nodesWithSufficientCpu
 }
+
+// This function fetches the cpus siblings associated with cpu core and returns them
+// Also updates the map by deleting the cpu siblings returned
+func GetCpuSiblings(numaCoreSiblings map[int]map[int][]int, coreKey int) []string {
+	var cpuSiblings []string
+	for key := range numaCoreSiblings {
+		for _, c := range numaCoreSiblings[key][coreKey] {
+			cpuSiblings = append(cpuSiblings, strconv.Itoa(c))
+			delete(numaCoreSiblings[key], c)
+		}
+	}
+	return cpuSiblings
+}
+
+//This function Splits the numa Siblings in to multiple Ranges
+//Example if we get core siblings as below:
+//[1 33 2 34 3 35 4 36 5 37 6 38 7 39 8 40 9 41 10 42],
+//the below function splits and returns ranges of 3 consecutive cpus:
+//1-3,4-6,7-9,33-35,36-38,39-41
+func GetRanges(cpuString string) string {
+	res := strings.Split(cpuString, ",")
```

Comment:
Suggestion (not a must) : use meaningful variable name. e.g:
 s/res/cpuList
 s/ret/cpuId
 s/cpusInt/cpuIds

Reply:
Addressed it in the latest patch